docs/developers/api.md

```diff
@@ -14,10 +14,12 @@ for binary compatiblity and aims at keeping source compatibility across
 versions.
 - Running a rule built against an older X or 0.Y version of Scalafix may cause
   either runtime errors, or false positive/negative tree selection. In that
-  case, a warning is issued when resolving the rules(s) artifact(s).
+  case, a warning calling the user to use a more recent version of the rule
+  or downgrading Scalafix is issued when resolving the rules(s) artifact(s).
 - Running a rule built against a more recent X.Y or 0.Y.Z version of Scalafix
-  is not possible. In that case, a `ScalafixException` is raised when resolving
-  the rules(s) artifact(s).
+  is also a potential source of incompatibility. In that case, a warning
+  calling the user to upgrade Scalafix is issued when resolving the rules(s)
+  artifact(s).
```

Comment:
I find this case easier to deal with than using a more recent rule.
The second requires rule author to publish a new version of their rule for each major version. In the other hand, we don't publish a major version very often!

Reply:
>  The second requires rule author to publish a new version of their rule for each major version

With this PR and the exception turned warning, it _encourages_ them (since they will get report from users complaining about warnings, as discussed in https://github.com/scalacenter/scalafix/pull/1588#issuecomment-1088347809) but does not _require_ them.